pkg/tsdb/cloudmonitoring/time_series_filter.go

```diff
@@ -30,54 +30,69 @@ func (timeSeriesFilter *cloudMonitoringTimeSeriesFilter) run(ctx context.Context
 		}
 		slog.Info("No project name set on query, using project name from datasource", "projectName", projectName)
 	}
-
 	r, err := s.createRequest(ctx, &dsInfo, path.Join("/v3/projects", projectName, "timeSeries"), nil)
 	if err != nil {
 		dr.Error = err
 		return dr, cloudMonitoringResponse{}, "", nil
 	}
 
-	r.URL.RawQuery = timeSeriesFilter.Params.Encode()
-	alignmentPeriod, ok := r.URL.Query()["aggregation.alignmentPeriod"]
+	first := true
+	d := cloudMonitoringResponse{}
 
-	if ok {
-		seconds, err := strconv.ParseInt(alignmentPeriodRe.FindString(alignmentPeriod[0]), 10, 64)
-		if err == nil {
-			if len(dr.Frames) == 0 {
-				dr.Frames = append(dr.Frames, data.NewFrame(""))
-			}
-			firstFrame := dr.Frames[0]
-			if firstFrame.Meta == nil {
-				firstFrame.SetMeta(&data.FrameMeta{
-					Custom: map[string]interface{}{
-						"alignmentPeriod": seconds,
-					},
-				})
+	for first || d.NextPageToken != "" {
```

Comment:
nit: a bit of a style change, you can move all the logic to do a request to a function so this code is cleaner (I'm not including params or errors):

```go
d := doRequest(...)
nextPageToken := d.NextPageToken
for nextPageToken != "" {
  nextPage := doRequest(..., nextPageToken)
  nextPageToken := nextPage.NextPageToken
  d.TimeSeries = append(d.TimeSeries, nextPage.TimeSeries...)
}

Reply:
I was going to add the same comment! :joy:

I know you didn't necessarily author this entire block but this section could be cleaned up to left-align "the happy path" as explained [here][mat] and [here][go wiki].

[go wiki]: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
[mat]: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
Neat! Thanks both, I'll make the changes

Reply:
I've extracted the function , however I have issues with left aligning the happy path @asimpson : I don't see how to do it since there is no possible return in `if` and no `else`. Any idea? 🤔